ctsit/nacculator
# Minor NP and FTLD fixes

This change adds/updates a few fields in the forms.py and builder.py files. It corrects some blanking logic in the np11 module forms.py file, and adds two FTLD-related fields to the Z1X that were missing in the IVP and FVP builder.py files. These fields already existed in the forms.py file, they had just been inaccessible without also appearing in builder.py.

This change was made so that NACCulator correctly catches errors in NP processing and stops throwing a false positive for the NPHEMO fields. It was also made so that PTIDs with FTLD data can be properly submitted to NACC without any missing data.

I also added the packet's visit number to the error log output for easier identification of issues in REDCap.


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. Checkout this branch and make sure you have a valid nacculator_cfg.ini file with a REDCap API token for the 1Florida ADRC REDCap project.  
2. From the command-line, run `nacculator_filters nacculator_cfg.ini` to download the latest REDCap data.
3. Next, run `redcap2nacc -ivp -ftld <run_$CURRENT-DATE/final_Update.csv >run_$CURRENT-DATE/fti_nacc_complete.txt 2>run_$CURRENT-DATE/fti_errors.txt`
4. The output should look like this: Regular NACCulator output. Upon submission to NACC, you will not receive any errors in the report about missing FTLD fields in the Z1X. In the errors.txt file, you will see the packet's visit number printed next to the PTID.
5. Next, export the data from the Brief Data Set project.
6. Run `redcap2nacc -np </path/to/np11/data.csv >np_nacc_complete.txt 2>np_errors.txt`
7. The output should look like this: In the error log, you will no longer see the error "#12b1 is '0' with length '1', but should be blank: 'Blank if Question 12b NPINF ne 1 (Yes)'." You may instead see "#12b1 is '0' with length '1', but should be blank: 'Blank if Question 12b NPHEMO ne 1 (Yes)'." if that error is present in the data, which is the correct error message.
8. If you pass _some bad value_, the output should look like this: regular NACCulator error logs.

- Verify the thing does this: Outputs correct blanking logic errors and Z1X forms.
- Verify the thing does not do that: Change any other behavior about NACCulator


## Affirmations

All of these should have a check by them. Any exception requires an explanation.
